README.md

```diff
@@ -61,3 +61,294 @@ julia> logpdf_with_trans(dist, x, false) # ignoring the transformation
 julia> logpdf_with_trans(dist, x, true) # considering the transformation
 -0.7760422307471244
 ```
+
+## `Bijector` interface
+Other than the `logpdf_with_trans` methods, the package also provides a more composable interface through the `Bijector` types. Consider for example the from above with `Beta(2, 2)`.
+
+```julia
+julia> using Bijectors; using Bijectors: Logit
+
+julia> dist = Beta(2, 2)
+Beta{Float64}(α=2.0, β=2.0)
+
+julia> x = rand(dist)
+0.7173326646959575
+
+julia> b = bijector(dist) # bijection (0, 1) → ℝ
+Logit{Float64}(0.0, 1.0)
+
+julia> y = b(x)
+0.9312689879144197
+```
+
+In this case we see that `bijector(d::Distribution)` returns the corresponding constrained-to-unconstrained bijection for `Beta`, which indeed is a `Logit` with `a = 0.0` and `b = 1.0`. The resulting `Logit <: Bijector` has a method `(b::Logit)(x)` defined, allowing us to call it just like any other function. Comparing with the above example, `b(x) == link(dist, x)`. Just to convince ourselves:
+
+```julia
+julia> b(x) == link(dist, x)
+true
+```
+
+What about `invlink`?
+
+```julia
+julia> b⁻¹ = inv(b)
+Inversed{Logit{Float64}}(Logit{Float64}(0.0, 1.0))
+
+julia> b⁻¹(y)
+0.7173326646959575
+
+julia> b⁻¹(y) == invlink(dist, y)
+true
+```
+
+Pretty neat, huh? `Inversed{Logit}` is also a `Bijector` where we've defined `(ib::Inversed{<:Logit})(y)` as the inverse transformation of `(b::Logit)(x)`. Note that it's not always the case that `inv(b) isa Inversed`, e.g. the inverse of `Exp` is simply `Log` so `inv(Exp()) isa Log` is true. Aslo, we can _compose_ bijectors:
+
+```julia
+julia> id = (b ∘ b⁻¹)
+Composed{Tuple{Inversed{Logit{Float64}},Logit{Float64}}}((Inversed{Logit{Float64}}(Logit{Float64}(0.0, 1.0)), Logit{Float64}(0.0, 1.0)))
+
+julia> id(y) == y
+true
+```
+
+This far we've seen that we can replicate the functionality provided by `link` and `invlink`. To replicate `logpdf_with_trans` we instead provide a `TransformedDistribution <: Distribution` implementing the `Distribution` interface from Distributions.jl:
+
+```julia
+julia> using Bijectors: TransformedDistribution
+
+julia> td = transformed(dist)
+TransformedDistribution{Beta{Float64},Logit{Float64},Univariate}(
+dist: Beta{Float64}(α=2.0, β=2.0)
+transform: Logit{Float64}(0.0, 1.0)
+)
+
+julia> td isa UnivariateDistribution
+true
+
+julia> logpdf(td, y)
```

Comment:
Nice - we can probably replace `logpdf_with_trans` with `logpdf(td::TransformedDistribution, y)` in Turing.

Reply:
Should be possible, thought might require some work? I'm not too familiar with exactly how the transformations happen in Turing, but just had a quick look and was not quite clear to me how to proceed in an efficient way. Using this interface I think the best way would be if we could somehow replace the original distributions with `TransformedDistribution` when we want to perform transformations, and then in `assume` return `inv(transformed_dist.transform)(y), logpdf(transformed_dist, y)`. As long as everything is type-stable, this should be efficient.

I've made an effort to make things in this package type-stable, but there's one `bijector(d::Distribution)` call that is currently _not_ type-stable: `Truncated`. See Issue #38. I wouldn't think this would pose much of an issue for application in Turing.jl since the construction of the distribution will occur in the very first construction of `VarInfo`, whose construction is already unstable.

Reply:
I think it's a good goal to have. One of the things I've had a lot of trouble with while working on Turing's sampler internals was when to link and when not to link. If we just had a bunch of `VarName => TransformedDistribution` pairs, we could theoretically handle all the gradients and everything automatically without anyone ever having to explicitly call `link!(vi, spl)` and `invlink!(vi, spl)`. We could also make sure that it's done in a performant way, so that transformations only occur when they absolutely need to.

Reply:
Oh yeah, I'm all for making the transition; I just wanted to say that it's probably not going to be as easy as _I_ thought at first:) You guys might have already been aware that's not as straight forward.